### PR TITLE
Update jms doc to add some usages about IBM MQ Client

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions/jms/runtime/src/main/doc/usage.adoc [hunk 1]
```diff
@@ -24,28 +24,93 @@ quarkus.pooled-jms.max-connections = 8
...
+        throw new RuntimeException("Unable to create new IBM MQ connection factory", e);
+    }
+    return wrapper.wrapConnectionFactory(mq);
+}
 ----
 
 [NOTE]
 ====
-`clientID` and `durableSubscriptionName` are not supported in pooling connections. If `setClientID` is called on a `reused` connection from the pool, an `IllegalStateException` will be thrown. You will get some error messages such like `Cause: setClientID can only be called directly after the connection is created`
+If you use `ibmmq-client` to consumer messages and enable XA, you needs to config `TransactionManager` in the camel router like this:
```
Comment (jamesnetherton) [inline extensions/jms/runtime/src/main/doc/usage.adoc:84]:
```suggestion
If you use `ibmmq-client` to consume messages and enable XA, you need to configure `TransactionManager` in the camel route like this:
```

File: extensions/jms/runtime/src/main/doc/usage.adoc [hunk 2]
```diff
@@ -24,28 +24,93 @@ quarkus.pooled-jms.max-connections = 8
...
+----
+@Inject
+TransactionManager transactionManager;
+
+@Override
+public void configure() throws Exception {
+    from("jms:queue:DEV.QUEUE.XA?transactionManager=#jtaTransactionManager");
+}
+
+@BindToRegistry("jtaTransactionManager")
```
Comment (vkasala) [inline extensions/jms/runtime/src/main/doc/usage.adoc:95]: @jamesnetherton @zhfeng 

Isn't it @Produces preffered over @BindToRegistry?

**Review by aldettinger — APPROVED** [review]
Beyond typos and BindToRegistry question, it looks good :+1:

File: extensions/jms/runtime/src/main/doc/usage.adoc (hunk 2 above)
Comment (zhfeng) [reply extensions/jms/runtime/src/main/doc/usage.adoc:95]: OK, let me check it.

Comment (zhfeng) [reply extensions/jms/runtime/src/main/doc/usage.adoc]: It should be good to use `@Named`.

**Pull request merged**